sympy/combinatorics/coset_table.py

```diff
@@ -766,14 +769,152 @@ def conjugates(self, R):
             R_set.difference_update(r)
         return R_c_list
 
+    ##############################
+    #      Modified Methods      #
+    ##############################
+
+    def modified_define(self, alpha, x):
+        '''
+        Input:  ∈ , x∈A, with  x undefined.
+        '''
+        len_table = len(self.table)
+        if len_table >= self.coset_table_max_limit:
+            # abort the further generation of cosets
+            raise ValueError("the coset enumeration has defined more than %s cosets. Try with a greater value max number of cosets "%C.coset_table_max_limit)
+        self.table.append([None]*len(self.A))
+        H = self.subgroup
+        self._grp = (list(free_group(', ' .join(["a_%d" % i for i in range(len(H))])))).pop(0)
+        self.P.append([None]*len(self.A))
+        beta = len_table
+        self.p.append(beta)
+        # Define alpha^x = beta, beta^(x**-1) = alpha
+        self.table[alpha][self.A_dict[x]] = beta
+        self.table[beta][self.A_dict_inv[x]] = alpha
+        # P[alpha][x] = epsilon, P[beta][x**-1] = epsilon
+        self.P[alpha][self.A_dict[x]] = self._grp.identity
+        self.P[beta][self.A_dict_inv[x]] = self._grp.identity
```

Comment:
This looks sufficiently like `define` that I'd give `define` a `modified` keyword. You can still define
```
def modified_define:
    define(..., modified=True)
```
if you think using `modified_define` will be more convenient.

Reply:
I was actually thinking of doing this. Will check it for the other methods too 👍

Reply:
This can still be merged with `define`.